localstack/services/kinesis/kinesis_starter.py

```diff
@@ -1,214 +1,80 @@
-        "START_STREAM_ENCRYPTION_DURATION={l} STOP_STREAM_ENCRYPTION_DURATION={l} "
-        "DEREGISTER_STREAM_CONSUMER_DURATION={l} MERGE_SHARDS_DURATION={l} SPLIT_SHARD_DURATION={l} "
-        "UPDATE_SHARD_COUNT_DURATION={l}"
-    ).format(l=latency)
-
-    if config.KINESIS_INITIALIZE_STREAMS != "":
-        initialize_streams_param = "INITIALIZE_STREAMS=%s" % config.KINESIS_INITIALIZE_STREAMS
-    else:
-        initialize_streams_param = ""
-
-    if kinesis_mock_bin.endswith(".jar"):
-        cmd = "KINESIS_MOCK_PLAIN_PORT=%s SHARD_LIMIT=%s %s %s %s %s java -XX:+UseG1GC -jar %s" % (
-            backend_port,
-            config.KINESIS_SHARD_LIMIT,
-            latency_param,
-            kinesis_data_dir_param,
-            log_level_param,
-            initialize_streams_param,
-            kinesis_mock_bin,
-        )
-    else:
-        chmod_r(kinesis_mock_bin, 0o777)
-        cmd = "KINESIS_MOCK_PLAIN_PORT=%s SHARD_LIMIT=%s %s %s %s %s %s --gc=G1" % (
-            backend_port,
-            config.KINESIS_SHARD_LIMIT,
-            latency_param,
-            kinesis_data_dir_param,
-            log_level_param,
-            initialize_streams_param,
-            kinesis_mock_bin,
-        )
-
-    return _run_proxy_and_command(
-        cmd=cmd,
-        port=port,
-        backend_port=backend_port,
-        update_listener=update_listener,
-        asynchronous=asynchronous,
-    )
+            raise ValueError('Unsupported Kinesis provider "%s"' % config.KINESIS_PROVIDER)
+    _start_kinesis_helper(_server, port=port, update_listener=update_listener)
+    return _server
 
 
-def start_kinesalite(port=None, asynchronous=False, update_listener=None):
-    # install and apply patches
-    install.install_kinesalite()
-    apply_patches_kinesalite()
-    # start up process
-    backend_port = get_free_tcp_port()
-    global PORT_KINESIS_BACKEND
-    PORT_KINESIS_BACKEND = backend_port
-    latency = config.KINESIS_LATENCY
-    kinesis_data_dir_param = ""
-    if config.dirs.data:
-        kinesis_data_dir = "%s/kinesis" % config.dirs.data
-        mkdir(kinesis_data_dir)
-        kinesis_data_dir_param = "--path %s" % kinesis_data_dir
-    cmd = (
-        "%s/node_modules/kinesalite/cli.js --shardLimit %s --port %s"
-        " --createStreamMs %s --deleteStreamMs %s --updateStreamMs %s %s"
-    ) % (
-        MODULE_MAIN_PATH,
-        config.KINESIS_SHARD_LIMIT,
-        backend_port,
-        latency,
-        latency,
-        latency,
-        kinesis_data_dir_param,
-    )
-
-    return _run_proxy_and_command(
-        cmd=cmd,
-        port=port,
-        backend_port=backend_port,
+def _start_kinesis_helper(server: Server, port=None, update_listener=None) -> Server:
+    server.start()
+    log_startup_message("Kinesis")
+    port = port or config.PORT_KINESIS
+    start_proxy_for_service(
+        "kinesis",
+        port,
+        backend_port=_server.port,
```

Comment:
Nitpicking: Since the function is only used in one other function and only consists of a few lines, I think we could just include that in the `start_kinesis` function directly.